test-tests/shared/src/test/scala/zio/test/laws/Laws2Spec.scala

```diff
@@ -0,0 +1,74 @@
+package zio.test.laws
+
+import zio.test._
+
+object Laws2Spec extends ZIOBaseSpec {
+
+  def equalTo[A: Equal](expected: A): Assertion[A] =
+    Assertion.assertion("equalTo")(Assertion.Render.param(expected))(_ === expected)
+
+  implicit class AssertEqualToSyntax[A](private val self: A) extends AnyVal {
+    def <->(that: A)(implicit eq: Equal[A]): TestResult =
+      assert(self)(equalTo(that))
+  }
+
+  trait Equal[-A] {
+    def equal(a1: A, a2: A): Boolean
+  }
+
+  object Equal {
+    implicit val byteEqual: Equal[Byte] = _ == _
+    implicit val charEqual: Equal[Char] = _ == _
+    implicit def listEqual[A: Equal]: Equal[List[A]] =
+      (l, r) => l.corresponds(r)(_ === _)
+    implicit val stringEqual: Equal[String] = _ == _
+    implicit def vectorEqual[A: Equal]: Equal[Vector[A]] =
+      (l, r) => l.corresponds(r)(_ === _)
+  }
+
+  implicit class EqualSyntax[A](private val self: A) extends AnyVal {
+    def ===(that: A)(implicit eq: Equal[A]): Boolean =
+      eq.equal(self, that)
+  }
+
+  final case class Equivalence[A, B](to: A => B, from: B => A)
+
+  object Equivalence extends Lawful2[Equivalence, Equal, Equal] {
+
+    val leftIdentity = new Laws2.Law1Left[Equivalence, Equal, Equal]("leftIdentity") {
```

Comment:
@jdegoes We are requiring a little more than we need to here, since technically the left law only needs an `Equal` instance for the left type and vice versa for the right law. Conceptually it would be nice to have something like the identity or the empty capability that provides no functionality and is always available. This would be the equivalent of `Any` for the environment except that `Any` is of the wrong kind.

I tried something like:

```scala
trait Identity[A]

object Identity {
  implicit def identity[A]: Identity[A] =
    new Identity[A] {}
}
```

But the problem is Scala can't unify that with other capabilities so it ends up leaking into the type signatures which is not ideal. Is there a way to express that? The type `AnyF[_]` such that for all `F[_]`, `F[x] <: AnyF[x]`?

Reply:
```scala
type AnyF[_] = Any
```

Reply:
That was easy! 😆